File: admin/lib/dto/user/user_response.dart
import 'package:admin/dto/user/user_get_dto.dart';

class UserResponse {
  final UserGetDto user;
  final int balance;

  UserResponse({required this.user, required this.balance});

  factory UserResponse.fromJson(Map<String, dynamic> json) => UserResponse(
    user: UserGetDto.fromJson(json['user']),
    balance: json['balance'],
  );
}

File: be/src/main/java/com/example/petclinic/dto/response/admin/user/UserResponse.java
package com.example.petclinic.dto.response.admin.user;

import com.example.petclinic.entity.User;
import lombok.AccessLevel;
import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.experimental.FieldDefaults;

@Data
@FieldDefaults(level = AccessLevel.PRIVATE, makeFinal = true)
@AllArgsConstructor
public class UserResponse {
    User user;
    int balance;
}